superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -362,176 +349,163 @@ class SqlEditor extends React.PureComponent {
+      dispatch(postStopQuery(latestQuery));
     }
-  }
+  };
 
-  createTableAs() {
-    this.startQuery(true, CtasEnum.TABLE);
-    this.setState({ showCreateAsModal: false, ctas: '' });
-  }
+  const createTableAs = () => {
+    startQuery(true, CtasEnum.TABLE);
+    setShowCreateAsModal(false);
+    setCtas('');
+  };
 
-  createViewAs() {
-    this.startQuery(true, CtasEnum.VIEW);
-    this.setState({ showCreateAsModal: false, ctas: '' });
-  }
+  const createViewAs = () => {
+    startQuery(true, CtasEnum.VIEW);
+    setShowCreateAsModal(false);
+    setCtas('');
+  };
 
-  ctasChanged(event) {
-    this.setState({ ctas: event.target.value });
-  }
+  const ctasChanged = event => {
+    setCtas(event.target.value);
+  };
 
-  queryPane() {
-    const hotkeys = this.getHotkeyConfig();
+  const queryPane = () => {
+    const hotkeys = getHotkeyConfig();
     const { aceEditorHeight, southPaneHeight } =
-      this.getAceEditorAndSouthPaneHeights(
-        this.state.height,
-        this.state.northPercent,
-        this.state.southPercent,
-      );
+      getAceEditorAndSouthPaneHeights(height, northPercent, southPercent);
     return (
       <Split
         expandToMin
         className="queryPane"
-        sizes={[this.state.northPercent, this.state.southPercent]}
-        elementStyle={this.elementStyle}
+        sizes={[northPercent, southPercent]}
+        elementStyle={elementStyle}
         minSize={200}
         direction="vertical"
         gutterSize={SQL_EDITOR_GUTTER_HEIGHT}
-        onDragStart={this.onResizeStart}
-        onDragEnd={this.onResizeEnd}
+        onDragStart={onResizeStart}
+        onDragEnd={onResizeEnd}
       >
-        <div ref={this.northPaneRef} className="north-pane">
+        <div ref={northPaneRef} className="north-pane">
           <AceEditorWrapper
-            actions={this.props.actions}
-            autocomplete={this.state.autocompleteEnabled}
-            onBlur={this.setQueryEditorSql}
-            onChange={this.onSqlChanged}
-            queryEditor={this.props.queryEditor}
-            database={this.props.database}
-            extendedTables={this.props.tables}
+            actions={actions}
+            autocomplete={autocompleteEnabled}
+            onBlur={setQueryEditorAndSaveSql}
+            onChange={onSqlChanged}
+            queryEditor={queryEditor}
+            database={database}
+            extendedTables={tables}
             height={`${aceEditorHeight}px`}
             hotkeys={hotkeys}
           />
-          {this.renderEditorBottomBar(hotkeys)}
+          {renderEditorBottomBar(hotkeys)}
         </div>
         <ConnectedSouthPane
-          editorQueries={this.props.editorQueries}
-          latestQueryId={this.props.latestQuery && this.props.latestQuery.id}
-          dataPreviewQueries={this.props.dataPreviewQueries}
-          actions={this.props.actions}
+          editorQueries={editorQueries}
+          latestQueryId={latestQuery && latestQuery.id}
```

Comment:
```suggestion
          latestQueryId={latestQuery?.id}
```
This can be shortened with optional chaining